dev/io.openliberty.microprofile.reactive.messaging.internal_fat/fat/src/io/openliberty/microprofile/reactive/messaging/fat/kafka/metrics/MetricsTest.java

```diff
@@ -0,0 +1,151 @@
+                                                                           MetricsReceptionBean.CHANNEL_MESSAGE_IN, APP_NAME, MetricsTestServlet.OUTGOING_TOPIC));
+
+        WebArchive war = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                        .addAsLibraries(kafkaClientLibs())
+                        .addAsManifestResource(kafkaPermissions(), "permissions.xml")
+                        .addPackage(MetricsTestServlet.class.getPackage())
+                        .addPackage(KafkaTestConstants.class.getPackage())
+                        .addPackage(AbstractKafkaTestServlet.class.getPackage())
+                        .addAsResource(appConfig, "META-INF/microprofile-config.properties");
+
+        ShrinkHelper.exportDropinAppToServer(server, war, SERVER_ONLY);
+        server.startServer();
+    }
+
+    @Test
+    public void testDeliverPayload() throws Exception {
+        // Test that metrics are updated when a payload is delivered via an emitter
+        runTest(server, APP_NAME + "/MetricsTest", "emitterDeliverPayload");
+        HashMap<String, Integer> metrics = getMetrics();
+
+        assertThat("5 messages not detected on channel", metrics.get("metrics-incoming") == 5);
+//      Assertion disabled as emitters currently don't report metrics per spec
+//        assertTrue("Outgoing and incoming channel counts are not the same",
+//                   metrics.get("mp_messaging_message_count_total{channel=\"MetricsEmitter\",mp_scope=\"base\",}") == metrics
+//                                   .get("mp_messaging_message_count_total{channel=\"metrics-incoming\",mp_scope=\"base\",}"));
+    }
+
+    @Test
+    public void testDeliverMessage() throws Exception {
+        // Test that metrics are updated when a message is delivered via delivery and reception beans
+        runTest(server, APP_NAME + "/MetricsTest", "deliverMessage");
+        HashMap<String, Integer> metrics = getMetrics();
+
+        assertThat("5 messages not detected on channel", metrics.get("metrics-outgoing") == 5);
+        assertThat("Outgoing and incoming channel counts are not the same",
+                   metrics.get("metrics-outgoing") == metrics
+                                   .get("metrics-message-incoming"));
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+        try {
+            server.stopServer();
+        } finally {
+            KafkaUtils.deleteKafkaTopics(KafkaTests.getAdminClient());
+        }
+    }
+
+    private HashMap<String, Integer> getMetrics() throws Exception {
+        // Returns the MicroProfile Reactive Messaging metrics from the /metrics endpoint
+        HttpURLConnection response = HttpUtils.getHttpConnection(new URL("http", server.getHostname(), 8010, "/metrics"),
+                                                                 200, null, 30, HTTPRequestMethod.GET, null, null);
```

Comment:
Using a hard-coded port is wrong here.

This should probably be changed to use methods on `componenttest.topology.utils.HttpUtils`